src/sentry/api/endpoints/oauth_userinfo.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework.authentication import get_authorization_header
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.models import ApiToken, UserEmail
+
+
+@control_silo_endpoint
+class OAuthUserInfoEndpoint(Endpoint):
+    authentication_classes = ()
+    permission_classes = ()
+
+    def get(self, request: Request) -> Response:
+        try:
+            access_token = get_authorization_header(request).split()[1].decode("utf-8")
+        except IndexError:
+            return Response("No access token found.", status=401)
+        try:
+            token_details = ApiToken.objects.get(token=access_token)
+        except ApiToken.DoesNotExist:
+            return Response(status=400)
+
+        scopes = token_details.get_scopes()
+        if "openid" not in scopes:
+            return Response(status=403)
+
+        user = token_details.user
+        user_output = {"sub": user.id}
+        if "profile" in scopes:
+            profile_details = {
+                "name": user.name,
+                "avatar_type": user.avatar_type,
+                "avatar_url": user.avatar_url,
+                "date_joined": user.date_joined,
```

Comment:
Generally our API outputs properties with camelBacked identifiers. Is there a reason to diverge from that convention here?

Reply:
yeah - the OAuth/OIDC requires snake case in order for this to be considered a valid implementation - [here's a thread discussing this exact thing](https://stackoverflow.com/questions/57831533/should-i-use-camelcase-for-oauth2).

However - this is only for the required OAuth stuff like `client_id`, `access_token` etc. So technically we could return those in snake case, and return the custom `userinfo` stuff in camelCase, but, I figured that would be even more confusing. LMK if you prefer it that way though and I can change it over!

Reply:
Fair point about the OIDC spec having standards for response keys. 🤔 Having two conventions in the same response is clunky. Lets go forward with what you have. Having consistency within the response which needs to follow the standard for some properties seems more important than our local API response key conventions.